src/interactions/potential.jl

```diff
@@ -0,0 +1,22 @@
+struct PotentialFunction{F}
+    f::F
+    # fwdiff gradient caches
+end
+
+function Molly.potential_energy(V::PotentialFunction,
+    sys,
+    neighbors=nothing;
+    n_threads=Threads.nthreads())
+
+    return V.f(sys.coords)
+end
+
+function Molly.forces(V::PotentialFunction,
+    sys,
+    neighbors=nothing;
+    n_threads=Threads.nthreads())
+
+    return ForwardDiff.gradient(V.f, sys.coords)
```

Comment:
This is currently not working because of the units (and maybe the Vector{SVector} as well).

ustrip_vec.() could help getting rid of the units, reinterpret with getting rid of the ArrayOfStructs (AoS).
But we would have to reconstruct this afterwards.
I have to say the units create quite a headache for me whenever I have to deal with them.
Maybe we need a convenient conversion interface between the combinations of AoS/Matrix/Vector with and without units?
In the above case something like
```
with_dropped_units(sys.coords, as=Vector, reconstruct=true) do x
  ForwardDiff.gradient(V.f, x)
 end
 ```
 would be great that would transform the coords to a single unitfree vector, call gradient and transform back to a unitful AoS.

Reply:
Switching to Zygote (of course...) resolved that problem.. Kind of. Zygote seems to struggle with reinterpret, that might be fixed by a gradient rule though..

I still would love to have a simple way to switch between units/no units and AoS/vec